dsl/frontend/policy.ml

```diff
@@ -0,0 +1,74 @@
+(* Changes to this type must also be reflected in `Ast.policy` in ast.ml *)
+type t =
+  | Class of Ast.clss
+  | Fifo of t list
+  | RoundRobin of t list
+  | Strict of t list
+  | WeightedFair of (t * float) list
+(* | EarliestDeadline of t list | ShortestJobNext of t list | ShortestRemaining
+   of t list | RateControlled of t list | LeakyBucket of t list * int * int |
+   TokenBucket of t list * int * int | StopAndGo of t list * int *)
+
+exception UnboundVariable of Ast.var
+exception UndeclaredClass of Ast.clss
+
+let lookup s x =
+  match List.assoc_opt x s with
+  | Some v -> v
+  | None -> raise (UnboundVariable x)
+
+let rec eval cl st (p : Ast.policy) =
+  (* Helper function that evaluates a policy list. *)
+  let eval_plst cl st = List.map (eval cl st) in
+
+  (* Helper function that evaluates a weighted policy list. *)
+  let eval_weighted_plst cl st = List.map (fun (x, i) -> (eval cl st x, i)) in
+
+  match p with
+  | Class c -> if List.mem c cl then Class c else raise (UndeclaredClass c)
+  | Var x -> eval cl st (lookup st x)
+  | Fifo plst -> Fifo (eval_plst cl st plst)
+  | RoundRobin plst -> RoundRobin (eval_plst cl st plst)
+  | Strict plst -> Strict (eval_plst cl st plst)
+  | WeightedFair wplst -> WeightedFair (eval_weighted_plst cl st wplst)
+  (* | EarliestDeadline plst -> EarliestDeadline (eval_plst cl st plst) |
+     ShortestJobNext plst -> ShortestJobNext (eval_plst cl st plst) |
+     ShortestRemaining plst -> ShortestRemaining (eval_plst cl st plst) |
+     RateControlled plst -> RateControlled (eval_plst cl st plst) | LeakyBucket
+     (plst, n1, n2) -> LeakyBucket (eval_plst cl st plst, n1, n2) | TokenBucket
+     (plst, n1, n2) -> TokenBucket (eval_plst cl st plst, n1, n2) | StopAndGo
+     (plst, n) -> StopAndGo (eval_plst cl st plst, n) *)
```

Comment:
I don't really want to drag this code around in comments. Please just stage a discussion or something, stick this code there, and remove it from here! Also applies elsewhere in this file ofc

Reply:
Done by [669f17b](https://github.com/cucapra/packet-scheduling/pull/57/commits/669f17be5e049e11412076887180458cb972121b) and issue #58